lib/ClangImporter/ImportType.cpp

```diff
@@ -1980,6 +1980,49 @@ ParameterList *ClangImporter::Implementation::importFunctionParameterList(
     ++index;
   }
 
+  auto genericParamTypeUsedInSignature =
+      [params, resultType](GenericTypeParamDecl *genericParam,
+                           bool shouldCheckResultType) -> bool {
+    auto paramDecl = genericParam->getClangDecl();
+    auto templateTypeParam = cast<clang::TemplateTypeParmDecl>(paramDecl);
+    // TODO: This won't work when we support importing dependent types. we'll
+    // have to change this logic to traverse the type tree of the imported
+    // Swift type (basically whatever ends up in the parameters variable).
+    // Check if genericParam's corresponding clang template type is used by
+    // the clang function's parameters.
+    for (auto param : params) {
+      if (hasSameUnderlyingType(param->getType().getTypePtr(),
+                                templateTypeParam)) {
+        return true;
+      }
+    }
+
+    // Check if genericParam is used as a type parameter in the result type.
+    return shouldCheckResultType &&
+        resultType.findIf([genericParam](Type typePart) -> bool {
+          return typePart->isEqual(genericParam->getDeclaredInterfaceType());
+        });
+  };
+  
+  // Make sure all generic parameters are accounted for in the function signature.
+  for (auto genericParam : genericParams) {
+    if (genericParamTypeUsedInSignature(genericParam, resultType->hasTypeParameter()))
```

Comment:
This currently crashes when importing `std` because `resultType` might be null. You can fix that by replacing `resultType->hasTypeParameter()` with `resultType && resultType->hasTypeParameter()`.

Reply:
Good catch. And this will also make it easier to add support for lazily loading result types.